webrecorder/browsertrix
# UX improvements to browser profile editing & creating view

Resolves https://github.com/webrecorder/browsertrix/issues/438



### Changes
- Layout and design pass on profile detail view to better match other detail views
- Ensures profile save buttons are always visible when creating or editing a profile
- Handle temporarily disconnected browser pings
- Enable recreating browser after it expires
- Show confirmation dialog when attempting to navigate away from active browser
- Tweak copy referencing profiles
- Refactors `profile-browser` into `TailwindComponent`

### Manual testing

1. Log in as crawler
2. Go to "Browser Profiles"
3. Click "New Browser Profile"
  a. Fill form
  b. Click "Configure Browser Profile". Verify "Interactive Browser" loads with website
  c. Scroll up and down page. Verify "Finish Browsing" control bar is visible while interactive browser is visible
  d. Open dev tools and switch to offline mode. Verify alert message is shown
  e. Wait until screen turns gray and then switch back to online mode. Verify "Reload" button is shown
  f. Click "Reload". Verify browser profile URL is reloaded
  g. Click "Finish Browsing" and save. Verify profile saves as expected
4. Click existing browser profile and repeat test steps 3c-3f.


### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Browser Profile Detail (updated copy) | <img width="1331" alt="Screenshot 2024-05-22 at 10 45 28 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/f1a46e29-aeb7-45fc-8b22-7ae73f43a983"> |
| Browser Profile Detail (updated action dropdown) | <img width="273" alt="Screenshot 2024-05-22 at 10 45 33 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/15e95679-b1ea-4ee2-be2c-27d1a6b2eed0"> |
| Browser Profile Detail - Editing (tooltip hover) | <img width="1335" alt="Screenshot 2024-05-22 at 10 50 31 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/7a35b988-62fa-43eb-a4c5-e3d94bff848a"> |
| Browser Profile Detail - Editing - Cancellation dialog | <img width="529" alt="Screenshot 2024-05-22 at 10 51 13 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/947b720a-5c78-45ca-8c33-8eaa355ee137"> |
| Browser Profile Detail - Editing (connection lost) | <img width="1320" alt="Screenshot 2024-05-22 at 10 58 35 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/72e706ef-b3da-4c46-adb0-f4e12e99d64f"> |
| Browser Profile Detail - Editing (browser expired) | <img width="1316" alt="Screenshot 2024-05-22 at 10 50 36 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/ff55b42f-663c-455b-946b-d9dd9ab5cc5a"> |
| New Browser Profile | <img width="1360" alt="Screenshot 2024-05-22 at 10 59 13 AM" src="https://github.com/webrecorder/browsertrix/assets/4672952/30bade00-729a-43ab-be42-bf69091cb676"> |

### Follow-ups

UX improvements to the browser profile detail layout and list view will be handled in https://github.com/webrecorder/browsertrix/issues/1816
